### PR TITLE
lora: rylrxxx: add missing timeout

### PR DESCRIPTION
**Pull request by JordanYates**
Add the missing timeout parameter to `modem_pipe_open`. 10 seconds appears to be the default value added to other in-tree drivers.

### CODE REVIEW

**Review by fabiobaltieri — APPROVED** [review]

**Review by kartben — APPROVED** [review]

**Review by rerickson1 — APPROVED** [review]

**Review by tomi-font — APPROVED** [review]

**Review by bjarki-andreasen — APPROVED** [review]

**Comment by tomi-font** [comment]
Made another PR that supersedes this one because two erroring `modem_pipe_close()` calls are left in this one: https://github.com/zephyrproject-rtos/zephyr/pull/76461

How come we didn't catch this in the offending PR, especially since it modified test and modem subsystem code?

**Review by tomi-font — CHANGES_REQUESTED** [review]
https://github.com/zephyrproject-rtos/zephyr/pull/76461

**Comment by bjarki-andreasen** [comment]
> Made another PR that supersedes this one because two erroring `modem_pipe_close()` calls are left in this one: #76461
> 
> How come we didn't catch this in the offending PR, especially since it modified test and modem subsystem code?

Because the merge window just opened :) PRs which have been ready for weeks/months ran on an old version of zephyr, and some PRs once merged will conflict with each other, even though they individually ran CI just fine. So unless we wait for CI to run after each PR is merged, we will get a delayed explosion :)

**Comment by tomi-font** [comment]
> Because the merge window just opened :) PRs which have been ready for weeks/months ran CI on an old version of zephyr, and some PRs once merged will conflict with each other, even though they individually ran CI just fine. So unless we wait for CI to run after each PR is merged, we will get a delayed explosion :)

Ah, indeed some other PR introduced those `modem_pipe_{open,close}()` calls. Well.

**Pull request closed without merging**
